src/client/main.cpp

```diff
@@ -1,12 +1,34 @@
 #include <iostream>
 
+#include "../common/bots/Bot.hpp"
 #include "../common/util/game_def.hpp"
 #include "../common/util/game_settings.hpp"
+#include "extern/cxxopts/include/cxxopts.hpp"
 #include "systems/ClientGameController.hpp"
 #include <SFML/Graphics.hpp>
 
-int main(int, char const**) {
-    ClientGameController client_game_controller;
-    client_game_controller.start();
-    return EXIT_SUCCESS;
+int main(int argc, char** argv) {
+    try {
+        cxxopts::Options options("Ungroup Client",
+                                 "The client for Ungroup - a game about temporary allainces.");
+        options.add_options()("x,headless", "Runs game without updating window.")(
+            "b,bot", "Runs game using a bot.")(
+            "s,strategy", "Runs game using specified strategy. Options are Random, NearestGreedy.",
+            cxxopts::value<int>());
+        auto result = options.parse(argc, argv);
+        bool is_headless = result["headless"].as<bool>();
+        bool is_bot = result["bot"].as<bool>();
+        bool strategy_opt_present = result.count("strategy");
+        BotStrategy strategy = BotStrategy::Random;
+        if (strategy_opt_present) {
+            strategy = (BotStrategy)result["strategy"].as<int>();
```

Comment:
Always use `static_cast<T>` over `(T)`
https://github.com/lefticus/cppbestpractices/blob/c4fb1cf8f8f0018c583437f79a9c05f3d263e99d/04-Considering_Safety.md#use-c-style-cast-instead-of-c-style-cast

in this case `static_cast<BotStrategy>(result["strategy"].as<int>());`

Reply:
good call didnt know about that